packages/react/src/firestore/useNamedQuery.ts

```diff
@@ -0,0 +1,35 @@
+import { useQuery, type UseQueryOptions } from "@tanstack/react-query";
+import {
+  type FirestoreError,
+  getDocs,
+  type Query,
+  type QuerySnapshot,
+  type DocumentData,
+} from "firebase/firestore";
+
+type FirestoreUseQueryOptions<TData = unknown, TError = Error> = Omit<
+  UseQueryOptions<TData, TError>,
+  "queryFn"
+> & {
+  firestore?: {
+    transform?: (snapshot: QuerySnapshot<DocumentData>) => TData;
+  };
+};
+
+export function useNamedQuery<TData = QuerySnapshot<DocumentData>>(
```

Comment:
I don't this hook is right... it should return a Query (or null) based on https://firebase.google.com/docs/reference/js/firestore_.md#namedquery_6438876

```
useNamedQuery(firestore, 'foo', { ... });
```